packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -100,22 +108,359 @@ class VMInfoList extends StatelessWidget {
       ],
     );
   }
+}
+
+List<Widget> _prettyRows(BuildContext context, List<Row> rows) {
+  return [
+    for (int i = 0; i < rows.length; ++i)
+      _buildAlternatingRow(context, i, rows[i]),
+  ];
+}
+
+Widget _buildAlternatingRow(BuildContext context, int index, Widget row) {
+  return Container(
+    color: alternatingColorForIndex(index, Theme.of(context).colorScheme),
+    height: defaultRowHeight,
+    padding: const EdgeInsets.symmetric(
+      horizontal: defaultSpacing,
+    ),
+    child: row,
+  );
+}
+
+/// A Refresh ToolbarAction button.
+class ToolbarRefresh extends ToolbarAction {
+  const ToolbarRefresh({
+    super.icon = Icons.refresh,
+    required super.onPressed,
+    super.tooltip = 'Refresh',
+  });
+}
+
+/// An IconLabelButton with label 'Request' and a 'call made' icon.
+class RequestDataButton extends IconLabelButton {
+  const RequestDataButton({
+    required super.onPressed,
+    super.icon = Icons.call_made,
+    super.label = 'Request',
+    super.outlined = false,
+  });
+}
+
+/// Displays a RequestDataButton if [requestedSize] is null, otherwise displays
+/// the requestable size and a ToolbarRefresh button next to it,
+/// to request that size again if required.
+class RequestableSizeWidget extends StatelessWidget {
+  const RequestableSizeWidget({
+    required this.requestedSize,
+    required this.requestFunction,
+  });
 
-  List<Widget> _prettyRows(BuildContext context, List<Row> rows) {
-    return [
-      for (int i = 0; i < rows.length; ++i)
-        _buildAlternatingRow(context, i, rows[i]),
-    ];
+  final InstanceRef? requestedSize;
+  final void Function()? requestFunction;
+
+  @override
+  Widget build(BuildContext context) {
+    if (requestedSize == null) {
+      return RequestDataButton(onPressed: requestFunction);
+    } else {
+      return Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          Text(
+            requestedSize!.valueAsString == null
+                ? '--'
+                : prettyPrintBytes(
+                    int.parse(requestedSize!.valueAsString!),
+                    includeUnit: true,
+                    kbFractionDigits: 1,
+                    maxBytes: 512,
+                  )!,
+          ),
+          ToolbarRefresh(onPressed: requestFunction),
+        ],
+      );
+    }
+  }
+}
+
+/// Wrapper to get the name of an ObjRef depending on its type.
+String? _objectName(ObjRef? objectRef) {
+  String? objectRefName;
+  if (objectRef == null) return null;
+
+  if (objectRef is ClassRef) objectRefName = objectRef.name;
```

Comment:
I'm guessing we can't just do this in a single conditional statement without needing a cast later?

Reply:
No, the compiler will not know all those types will have the 'name' field.